frontend/widgets/VolumeControl.hpp

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include <obs.hpp>
+
+#include <components/VolumeSlider.hpp>
+#include <widgets/VolumeName.hpp>
+
+#include <Idian/Utils.hpp>
+#include <QFrame>
+#include <QPushButton>
+#include <QWidget>
+
+class MuteCheckBox;
+class QBoxLayout;
+class QLabel;
+class VolumeMeter;
+
+namespace OBS {
+enum class MixerStatus : uint32_t {
+	None = 0,
+	Active = 1 << 0,
+	Locked = 1 << 1,
+	Global = 1 << 2,
+	Pinned = 1 << 3,
+	Hidden = 1 << 4,
+	Unassigned = 1 << 5,
+	Preview = 1 << 6,
+};
+
+inline MixerStatus operator|(MixerStatus a, MixerStatus b)
+{
+	return static_cast<MixerStatus>(static_cast<uint32_t>(a) | static_cast<uint32_t>(b));
+}
+
+inline MixerStatus operator&(MixerStatus a, MixerStatus b)
+{
+	return static_cast<MixerStatus>(static_cast<uint32_t>(a) & static_cast<uint32_t>(b));
+}
+
+inline MixerStatus operator~(MixerStatus a)
+{
+	return static_cast<MixerStatus>(~static_cast<uint32_t>(a));
+}
+} // namespace OBS
+
+class VolumeControl : public QFrame {
+	Q_OBJECT
+
+private:
+	std::unique_ptr<idian::Utils> utils;
+
+	OBSWeakSource weakSource_;
+	const char *uuid;
+	std::vector<OBSSignal> sigs;
```

Comment:
```suggestion
	std::vector<OBSSignal> signals;
```

Reply:
Qt has a macro named `signals`. Renamed to `obsSignals`